### PR TITLE
MM-47782 Clear unused values from local storage on logout

### PR DESCRIPTION
**Pull request by hmhealey**
My previous PR stopped us from using localStorage to cache values for colouring usernames in compact mode, but it still left those values in storage if they already existed there. This removes those on logout without clearing all of localStorage because other values in there may be used by other parts of the app.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-47782

#### Release Note
```release-note
Clean up unused values from browser storage on logout
```

### CODE REVIEW

File: actions/storage.js [hunk 1]
```diff
@@ -51,3 +51,13 @@ export function actionOnGlobalItemsWithPrefix(prefix, action) {
...
 }
+
+export function cleanLocalStorage() {
+    for (const key of Object.keys(localStorage)) {
+        // Remove all keys added for user profile colours before MM-47782
+        const userProfileColorPattern = /^[a-z0-9.\-_]+-#[a-z0-9]{6}$/i;
+        if (userProfileColorPattern.test(key)) {
+            localStorage.removeItem(key);
+        }
+    }
```
Comment (M-ZubairAhmed) [inline actions/storage.js:62]: One would thing by the function name [clears complete LC](https://developer.mozilla.org/en-US/docs/Web/API/Storage/clear) can we rename it to be more specific

File: actions/storage.js [hunk 2]
```diff
@@ -51,3 +51,13 @@ export function actionOnGlobalItemsWithPrefix(prefix, action) {
         data: {prefix, action},
     };
 }
+
+export function cleanLocalStorage() {
+    for (const key of Object.keys(localStorage)) {
+        // Remove all keys added for user profile colours before MM-47782
+        const userProfileColorPattern = /^[a-z0-9.\-_]+-#[a-z0-9]{6}$/i;
```
Comment (BenCookie95) [inline actions/storage.js:58]: Nit: move this outside the loop

File: actions/storage.js (hunk 1 above)
Comment (hmhealey) [reply actions/storage.js:62]: I can't come up with a better name for this that isn't overly descriptive, especially if we start removing other stuff from the localStorage on logout, so I think this is good enough.

**Review by BenCookie95 — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by nevyangelova — APPROVED** [review]

**Pull request merged**